### PR TITLE
Fix registration

### PR DESCRIPTION
**Pull request by Yaron-Jack**

### CODE REVIEW

File: app/(tabs)/Deposit.tsx [hunk 1]
```diff
@@ -82,21 +84,12 @@ export default function Deposit() {
               ...styles.amountLabel,
             }}
           >
-            { i18n.t('deposit_form_amount') }
+            {i18n.t('deposit_form_amount')}
```
Comment (GILREICH1) [inline app/(tabs)/Deposit.tsx:87]: I don't think this was changed on purpose? will cause formatting back and forth in future. Please undo

File: app/Auth.tsx [hunk 2]
```diff
@@ -119,27 +157,51 @@ export default function Auth() {
...
       <View style={{ padding: 8, flexDirection: 'row' }}>
         <CustomButton
           text={regUI ? i18n.t('auth_register') : i18n.t('auth_login')}
-          disabled={!phoneNumber || (regUI && (!firstName || !lastName))}
+          disabled={isRegButtonDisabled}
           onPress={onSubmit}
         />
       </View>
+      <Image
+        source={require('/Users/Lenovo/app/app/images/LiraShapiraLogo.jpeg')}
```
Comment (GILREICH1) [inline app/Auth.tsx:189]: Images in use by app need to be in assets directory

File: app/Auth.tsx [hunk 3]
```diff
@@ -23,22 +29,28 @@ import { CustomModal } from '../components/utils/CustomModal';
...
+
   const colorScheme = useColorScheme() ?? 'light';
   const dispatch = useAppDispatch();
   const router = useRouter();
   const [regUI, setRegUI] = useState<boolean>(true);
   const phoneNumber = useAppSelector(selectPhoneNumber);
   const firstName = useAppSelector(selectFirstName);
   const lastName = useAppSelector(selectLastName);
+  const selectedCompostStand = useAppSelector(selectCompostStand);
+  const isRegButtonDisabled = !phoneNumber || (regUI && (!firstName || !lastName)) || !selectedCompostStand;
```
Comment (GILREICH1) [inline app/Auth.tsx:41]: Maybe:
```suggestion
  const isRegButtonDisabled = !phoneNumber || (regUI && (!firstName || !lastName || !selectedCompostStand));
```
We only require the user to select a stand if they are registering. Is it possible the current format of the conditional will require users who are logging in to add a compost stand?

File: app/Auth.tsx [hunk 4]
```diff
@@ -103,10 +141,10 @@ export default function Auth() {
           />
         </View>
         {regUI ? (
-         <Registration />
+          <Registration />
         ) : null}
         <View style={{ flexDirection: 'row', alignItems: 'center' }}>
-          <Text style={{paddingRight: 8 }}>+972</Text>
+          <Text style={{ paddingRight: 8 }}>+972</Text>
```
Comment (GILREICH1) [inline app/Auth.tsx:147]: please reset

File: app/Auth.tsx [hunk 5]
```diff
@@ -119,27 +157,51 @@ export default function Auth() {
               height: 36,
               marginHorizontal: 1,
               width: '100%',
-              alignSelf: 'center', // Center the TextInput element horizontally
+              alignSelf: 'center',
             }}
-            placeholder={ i18n.t('auth_phone_number') }
+            placeholder={i18n.t('auth_phone_number')}
```
Comment (GILREICH1) [inline app/Auth.tsx:162]: reset

File: app/images/LiraShapiraLogo.jpeg
Comment (GILREICH1) [inline app/images/LiraShapiraLogo.jpeg:1]: please put images in assets

File: components/auth/Registration.tsx
Comment (GILREICH1) [inline components/auth/Registration.tsx:1]: looks like just indentation changed? please reset or clarify

File: translationService/he.ts [hunk 6]
```diff
@@ -40,6 +40,8 @@ const heDictionary: Dictionary = {
   deposit_form_notes: 'הערות / הארות',
   deposit_form_amount: 'כמות',
 
+  location: 'בחר מיקום מתפריט גלילה',
+  deposit_compost_stand_blank: ' ',
```
Comment (GILREICH1) [inline translationService/he.ts:44]: should be empty string

File: tsconfig.json [hunk 7]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (GILREICH1) [inline tsconfig.json:1]: undo please :)

File: types/Deposit.ts [hunk 8]
```diff
@@ -9,7 +10,7 @@ export enum CompostStand  {
   burma = 'burma',
   park_sonya = 'park_sonya',
   kerem_hazeitim = 'kerem_hazeitim',
-  hizkiyahu_hamelech = 'hizkiyahu_hamelech',
+  hizkiyahu_hamelech = 'hizkiyahu_haFmelech',
```
Comment (GILREICH1) [inline types/Deposit.ts:13]: undo please :)

File: app/Auth.tsx [hunk 9]
```diff
@@ -65,13 +80,36 @@ export default function Auth() {
...
   const onFailedLogin = () => {
     dispatch(setIsModalVisible(false));
     setRegUI(true);
   };
 
   return (
     <View style={styles.container}>
+      <View>
+        <View style={styles.headerContainer}>
+          <h1 style={styles.headerText}>{i18n.t('main_title')}</h1>
```
Comment (GILREICH1) [inline app/Auth.tsx:109]: I think we'll need to run this through our design committee. Please include screenshot in PR but remove text for now 👍

**Review by GILREICH1 — CHANGES_REQUESTED** [review]
Thanks for this work Jack! Really well integrated and much needed feature. 

I left comments on changes needed, a lot of which is extra indentations which will cause formatter conflicts.

Also, as you did not add or remove any npm package, I don't think package-lock or package.json changes should be made. please remove these from your PR 👍

Thank you, looking forward to seeing this in live :)

File: app/(tabs)/Deposit.tsx (hunk 1 above)
Comment (Yaron-Jack) [reply app/(tabs)/Deposit.tsx:87]: strange, in my branch it appears only once...

Screenshot 2023-11-29 at 15 35 30

Comment (Yaron-Jack) [reply app/(tabs)/Deposit.tsx:87]: I have automatic formatting so It just changes it like this every time I save(I just checked as well) It strange that it appears like a change on git

same for the other formatting issues
would you like me to cancel it? It just usually makes life easier..

File: app/Auth.tsx (hunk 9 above)
Comment (Yaron-Jack) [reply app/Auth.tsx:108]: Sure :)

Screenshot 2023-11-29 at 16 04 40

File: app/Auth.tsx (hunk 3 above)
Comment (Yaron-Jack) [reply app/Auth.tsx]: You're right :)

**Review by GILREICH1 — APPROVED** [review]

**Pull request merged**